### PR TITLE
Fixes #36591 - Fix the parameter value masking

### PR DESCRIPTION
**Pull request by girijaasoni**

### CODE REVIEW

**Comment by nofaralfasi** [comment]
@girijaasoni Your changes also affect the common_parameters page. The value column for parameters with hidden_value shows `true` instead of asterisks.

**Review by nofaralfasi — CHANGES_REQUESTED** [review]
@girijaasoni Your changes also affect the common_parameters page. The value column for parameters with hidden_value shows true instead of asterisks.

**Comment by girijaasoni** [comment]
> @girijaasoni Your changes also affect the common_parameters page. The value column for parameters with hidden_value shows true instead of asterisks.

That was because i changed the parameters model, fixed it now :)

**Comment by nofaralfasi** [comment]
It is working for me now. Just one small note, I think it would be nicer to hide the parameter's value if the `Hide value` option is enabled, as we do on the common_parameters page.

**Comment by girijaasoni** [comment]
> It is working for me now. Just one small note, I think it would be nicer to hide the parameter's value if the `Hide value` option is enabled, as we do on the common_parameters page.

as per the acceptance criteria in the bz: https://bugzilla.redhat.com/show_bug.cgi?id=2172394, the expected results are mentioned as "The value should be visible even after hide and unhide."

**Comment by nofaralfasi** [comment]
> > It is working for me now. Just one small note, I think it would be nicer to hide the parameter's value if the `Hide value` option is enabled, as we do on the common_parameters page.
> 
> as per the acceptance criteria in the bz: https://bugzilla.redhat.com/show_bug.cgi?id=2172394, the expected results are mentioned as "The value should be visible even after hide and unhide."

I believe their intention was that when you hide and unhide the value, you would see the original value again instead of asterisks.

**Comment by girijaasoni** [comment]
> > > It is working for me now. Just one small note, I think it would be nicer to hide the parameter's value if the `Hide value` option is enabled, as we do on the common_parameters page.
> > 
> > 
> > as per the acceptance criteria in the bz: https://bugzilla.redhat.com/show_bug.cgi?id=2172394, the expected results are mentioned as "The value should be visible even after hide and unhide."
> 
> I believe their intention was that when you hide and unhide the value, you would see the original value again instead of asterisks.

Thanks @nofaralfasi , fixed it :)

**Review by nofaralfasi — CHANGES_REQUESTED** [review]
If I hide a value and then proceed to input something, when I reveal it again, only the most recent character I typed is visible, and the original value is replaced with asterisks.

**Comment by nofaralfasi** [comment]
[screen-capture (3).webm](https://github.com/theforeman/foreman/assets/38184193/445e734e-8967-4130-9df8-6bbac8a6a2d0)

**Comment by girijaasoni** [comment]
> If I hide a value and then proceed to input something, when I reveal it again, only the most recent character I typed is visible, and the original value is replaced with asterisks.

fixed it but for all types of input except for the boolean type as it is a dropdown in that case and the component we have used is "Select" and it doesn't have a parameter for masking a value and IMHO it's not required  as well.

**Comment by stejskalleos** [comment]
ok to test

File: app/views/api/v2/parameters/base.json.rabl [hunk 1]
```diff
@@ -1,7 +1,3 @@
 object @parameter
 
-attributes :id, :name, :parameter_type, :associated_type, :hidden_value?
-
-node do
-  partial("api/v2/common/show_hidden", :locals => { :value => :value }, :object => @object)
```
Comment (stejskalleos) [inline app/views/api/v2/parameters/base.json.rabl]: What was the reason to remove this node? We should not change the API,
customers can have their own infra on the top pf our API and this is a breaking change

File: webpack/assets/javascripts/react_app/components/HostDetails/Tabs/Parameters/EditTableRow.js [hunk 2]
```diff
@@ -135,7 +134,7 @@ export const EditParametersTableRow = ({
               <SelectOption value="false" />
             </Select>
           ) : (
-            <TextArea
+            <TextInput
```
Comment (stejskalleos) [inline webpack/assets/javascripts/react_app/components/HostDetails/Tabs/Parameters/EditTableRow.js]: I don't think this is a good change. For short strings, yeah, but what if you'll have a parameter with let's say public SSH key, which is a common use case:
```
ssh-rsa AAAAB3NzaC1yc2EAAAADAQABAAABgQDLLfFKVRh3XWCZBhFZ9VTHk1dnjDAYuyPpGxXUdyaMBEA0mBk7xNW1vP3zikdo4rdn4lg4q1sLeh2MkkmZnBiV+AcetaQ3iceNfRKjCL7hML4fAyKqIC2FuO7zAFWbe1jqHDYz7TiOEIqmfJFCPnHHw2XyqKhLiMjsRqjLuTPeGeDmOOo1aI7KwTM1iyrrJsjKtuBT4navog5TNaTyR+E9KC+KPcTWSG61nA0HyOE/UD2jSVI+8Q13E6H2RfbMmbRzIJPUrxuS5B8Io2LxEr56YZuKeQhsYWJBqtpmvhKQULQo4WdPu+1nyApInXbF25jx0FieBROJ1D+KLl/Uj4Tp4O9ihPAwiUo3F6qK0aJFt01YvxEERFmqFdyI7+z/bOXlj0yO1qnFDphBd0s0uk/3FHcGZRqLtKXywv9aDbprjdNJzDAW6/j93n8yn1aaaaaaaaaaaaaaaaAAAAAAAAAAAAAAAAAAAAMgGLeQIAsWvU6PLwfHq8polj6xntR9XDv3/9mkNBgU= user@localhost.localdomain
```

That's not something you want to have in the `input` field.

**Review by stejskalleos — CHANGES_REQUESTED** [review]
If I understand it correctly, we are masking the value of the parameter while **editing** it, right?

My question is, what's the point of hiding value while editing it? If users have permission for editing parameters, then there is no reason to hide the value for them.
IMHO it's a security feature with low to no impact that can be removed, but please feel free to correct me if I miss something.

**Comment by adarshdubey-star** [comment]
> If I understand it correctly, we are masking the value of the parameter while **editing** it, right?
> 
> My question is, what's the point of hiding value while editing it? If users have permission for editing parameters, then there is no reason to hide the value for them. IMHO it's a security feature with low to no impact that can be removed, but please feel free to correct me if I miss something.

So the BZ was ideally filed for the scenario when you add a parameter and click on the hide button and again unclick it, the value vanished, in the context of editing a parameter the user can surely unmask and change the value and again mask it if it's a critical parameter based on the scenario.

**Review by stejskalleos — APPROVED** [review]
:green_apple: **LGTM**, fix the issue

**Comment by stejskalleos** [comment]
Thanks @girijaasoni @nofaralfasi

**Pull request merged**